packages/core/src/Page/PageVueServerRenderer.ts

```diff
@@ -3,49 +3,52 @@
   due to the usage of `module`. Hence, the use of the above `eslint-disable`.
 */
-import Vue from 'vue';
+import * as Vue from 'vue';
+import { createSSRApp } from 'vue';
+import { renderToString } from 'vue/server-renderer';
+import { compileTemplate } from 'vue/compiler-sfc';
+import type { SFCTemplateCompileOptions, CompilerOptions } from 'vue/compiler-sfc';
+
 import path from 'path';
 import fs from 'fs-extra';
-import * as VueCompiler from 'vue-template-compiler';
-import { createRenderer } from 'vue-server-renderer';
 import * as logger from '../utils/logger';
 import type { PageConfig, PageAssets } from './PageConfig';
 import type { Page } from '.';
-/* eslint-enable import/no-import-module-exports */
 
 let bundle = require('@markbind/core-web/dist/js/vueCommonAppFactory.min');
 
-const { renderToString } = createRenderer();
-
-interface PageEntry {
-  page: Page;
-  compiledVuePage: VueCompiler.CompiledResultFunctions;
-  pageNav: string;
-}
-
-// hold the mapping of sourcePath to latest built pages (for hot-reload dev purposes)
-const pageEntries: Record<string, PageEntry> = {};
-
 /**
- * Compiles page into Vue Application to get the page render function and places
- * it into a script so that the browser can retrieve the page render function to
- * render the page during Vue mounting.
+ * Compiles a Vue page template into a JavaScript function returning render function
+ * and saves it as a script file so that the browser can access to
+ * generate VDOM during Vue mounting and conduct patching.
  *
  * This is to avoid the overhead of compiling the page into Vue application
  * on the client's browser (alleviates FOUC). It is also the pre-requisite to enable SSR.
+ * See https://github.com/vuejs/core/blob/main/packages/compiler-core/src/options.ts for
+ * compilerOptionTypes
  *
  * @param content Page content to be compiled into Vue app
  */
-async function compileVuePageAndCreateScript(
-  content: string, pageConfig: PageConfig, pageAsset: PageAssets):
-  Promise<VueCompiler.CompiledResultFunctions> {
-  // Compile Vue Page
-  const compiledVuePage = VueCompiler.compileToFunctions(content);
-
-  // Set up script content
-  const outputContent = `
-    var pageVueRenderFn = ${compiledVuePage.render};
-    var pageVueStaticRenderFns = [${compiledVuePage.staticRenderFns}];
+async function compileVuePageCreateAndReturnScript(
+  content: string, pageConfig: PageConfig, pageAsset: PageAssets) {
+  const compilerOptions: CompilerOptions = {
+    runtimeModuleName: 'vue',
+    runtimeGlobalName: 'Vue',
+    mode: 'function',
+  };
+
+  const templateOptions: SFCTemplateCompileOptions = {
+    source: content,
+    filename: pageConfig.sourcePath,
+    id: pageConfig.sourcePath,
+    compilerOptions,
+  };
+  const compiled = compileTemplate(templateOptions);
+  const outputContent = compiled.code;
+
+  const scriptContent = `
+    const renderFn = new Function(${JSON.stringify(outputContent)});
+    var render = renderFn();
```

Comment:
This part looks good to me. I think is a very good implementation of what we really have.